action/builder.go

```diff
@@ -159,16 +169,39 @@ func (b *EnvelopeBuilder) BuildTransfer(tx *types.Transaction) (Envelope, error)
 	if tx.To() == nil {
 		return nil, ErrInvalidAct
 	}
-	b.setEnvelopeCommonFields(tx)
+	if err := b.setEnvelopeCommonFields(tx); err != nil {
+		return nil, err
+	}
 	b.elp.payload = NewTransfer(tx.Value(), getRecipientAddr(tx.To()), tx.Data())
 	return b.build(), nil
 }
 
-func (b *EnvelopeBuilder) setEnvelopeCommonFields(tx *types.Transaction) {
-	b.elp.nonce = tx.Nonce()
-	b.elp.gasPrice = new(big.Int).Set(tx.GasPrice())
-	b.elp.gasLimit = tx.Gas()
-	b.elp.accessList = tx.AccessList()
+func (b *EnvelopeBuilder) setEnvelopeCommonFields(tx *types.Transaction) error {
+	v, err := convertEthTxType(tx.Type())
+	if err != nil {
+		return err
+	}
+	b.ab.version = uint32(v)
+	b.ab.nonce = tx.Nonce()
+	b.ab.gasPrice = tx.GasPrice()
+	b.ab.gasLimit = tx.Gas()
+	b.ab.accessList = tx.AccessList()
+	b.ab.gasFeeCap = tx.GasFeeCap()
+	b.ab.gasTipCap = tx.GasTipCap()
+	return nil
+}
+
+func convertEthTxType(typ uint8) (int, error) {
+	switch typ {
+	case types.LegacyTxType:
+		return LegacyTxType, nil
+	case types.AccessListTxType:
+		return AccessListTxType, nil
+	case types.DynamicFeeTxType:
+		return DynamicFeeTxType, nil
+	default:
+		return 0, errors.Wrapf(ErrInvalidAct, "unsupported eth tx type %d", typ)
+	}
```

Comment:
we can actually deprecate `setEnvelopeCommonFields`, instead:
```
func (b *EnvelopeBuilder) convertEthTx(tx *types.Transaction) (TxCommonWithProto, error) {
    switch tx.Type() {
    case types.LegacyTxType:
        return LegacyTx{...}
    }
}
```

Reply:
will refactor builder in next PR